src/Compilers/CSharp/Test/Semantic/Semantics/PropertyFieldKeywordTests.cs

```diff
@@ -0,0 +1,1042 @@
+        [InlineData("set => field = value; get { return 0; }")] // expression bodied setter
+        public void TestSemanticModel_GetterAndSetter_OnlyOneAccessorUsesKeyword(string accessor)
+        {
+            var comp = CreateCompilation($@"
+public class C
+{{
+    public int P1 {{ {accessor} }}
+}}
+");
+            comp.VerifyDiagnostics();
+            var tree = comp.SyntaxTrees[0];
+            var model = comp.GetSemanticModel(tree);
+
+            var fieldToken = tree.GetRoot().DescendantTokens().Where(t => t.ContextualKind() == SyntaxKind.FieldKeyword).Single();
+            AssertEx.NotNull(fieldToken.Parent);
+            var info = model.GetSymbolInfo(fieldToken.Parent);
+            Assert.Empty(info.CandidateSymbols);
+            Assert.False(info.IsEmpty);
+            Assert.Equal("<P1>k__BackingField", info.Symbol!.Name);
+        }
+
+        [Fact]
+        public void TestGetFieldsToEmit()
+        {
+            var comp = CreateCompilation(@"
+public class C
+{
+    public string P { set => field = value; }
+}
+");
+            for (int i = 0; i < 3; i++)
+            {
+                var fields = ((SourceMemberContainerTypeSymbol)comp.GetTypeByMetadataName("C")!).GetFieldsToEmit().ToArray();
+                Assert.Equal(1, fields.Length);
+                Assert.Equal("<P>k__BackingField", fields[0].Name);
+            }
+        }
+
+        [Fact]
+        public void TestGetFieldsToEmit_ExpressionBodied()
+        {
+            var comp = CreateCompilation(@"
+public class C
+{
+    public string P => field;
+}
+");
+            for (int i = 0; i < 3; i++)
+            {
+                var fields = ((SourceMemberContainerTypeSymbol)comp.GetTypeByMetadataName("C")!).GetFieldsToEmit().ToArray();
+                Assert.Equal(1, fields.Length);
+                Assert.Equal("<P>k__BackingField", fields[0].Name);
+            }
+        }
+
+        [Fact]
+        public void TestGetFieldsToEmit_Initializer()
+        {
+            var comp = CreateCompilation(@"
+public class C
+{
+    public string P { get; } = string.Empty;
+}
+");
+            for (int i = 0; i < 3; i++)
+            {
+                var fields = ((SourceMemberContainerTypeSymbol)comp.GetTypeByMetadataName("C")!).GetFieldsToEmit().ToArray();
+                Assert.Equal(1, fields.Length);
+                Assert.Equal("<P>k__BackingField", fields[0].Name);
+            }
+        }
+
+        [Fact]
+        public void AssignReadOnlyOnlyPropertyOutsideConstructor()
+        {
+            var comp = CreateCompilation(@"
+class Test
+{
+    int X
+    {
+        get
+        {
+            field = 3;
+            X = 3;
+            return 0;
+        }
+    }
+}
+");
+            comp.VerifyDiagnostics(
+                // (8,13): error CS0191: A readonly field cannot be assigned to (except in a constructor or init-only setter of the type in which the field is defined or a variable initializer)
```

Comment:
>// (8,13): error CS0191: A readonly field cannot be assigned to (except in a constructor or init-only setter of the type in which the field is defined or a variable initializer) [](http://example.com/codeflow?start=16&length=177)

This error doesn't make sense to me. I understand that the spec requires this field to be read-only, but I don't think this restriction is justified. Let's add a PROTOTYPE comment and record the design issue. I am not sure where you record them. In the speclet? Elsewhere?

Reply:
@AlekseyTs Do you expect another error? What exactly?


> I am not sure where you record them. In the speclet? Elsewhere?

I was recording things in a local text file, but started to move everything to PROTOTYPE comments for easier tracking.

The list I've locally is:

- Test _isAutoPropertyAccessor (ERR_AutoSetterCantBeReadOnly, IsDeclaredReadOnly)
    - I already added tests for ERR_AutoSetterCantBeReadOnly, and it seems working as expected.
- Review callers of BackingField (AbstractFlowPass, DefiniteAssignment, LocalRewriter)
- Review callers of IsAutoProperty & IsAutoPropertyWithGetAccessor
- Review callers of AssociatedField (for both props and events)
- Test "field" attribute target.
  
- Test `diagnostics.Add(ErrorCode.ERR_PartialMethodReadOnlyDifference, implementation.Locations[0]);` code path (Depends on accessor's IsDeclaredReadOnly)
- `isAutoProperty = isAutoProperty && !(containingType.IsInterface && !IsStatic) && !IsAbstract && !IsExtern && !isIndexer;`  (Test this for field keyword)
- Field created during binding shouldn't be emitted for speculative semantic model
- AllowedAttributeLocations for set; only property.
- Spec disallows semicolon-only `set` accessor without a `get` accessor. But allows `public string S { set => field = value; }`. Add tests for this scenario.
- Add tests for init accessors
- Review PropertyTests.cs for any interesting tests.
- Test for
    ```csharp
    public int P
    {
        get
        {
            int field = 5;
            return field;
        }
        set
        {
            field = value;
        }
    }
    ```

- In a situation when we don't have a semicolon-only accessor, but have an initializer, we should create
   a backing field with IsCreatedForFieldKeyword flag optimistically set to true. BTW, I am fine with IsCreatedForFieldKeyword == true in this error scenario as apposed
   to IsCreatedForFieldKeyword == false in presence of the field keyword and absence of a semicolon-only accessor, which is a success scenario.
   Later on, we obviously need to confirm that there is indeed a field keyword usage and report an error if there isn't one. This is
   the part that requires additional thinking to figure out where is the right place to perform the check and what would be the mechanism to
   confirm the actual presence of the field keyword. Right now I am leaning towards recording this fact as a separate
   flag in SynthesizedBackingFieldSymbol (there is still question when to do that). It is very likely that this flag is going to be
   required anyway, because we want to be able to distinguish SynthesizedBackingFieldSymbols that were created from speculative
   binding through SemanticModel and those that correspond to "real" field keywords in the original source.

- Should initializers use backing field or setter? Also consider this `public int P { get => field; } = 5;`
- Regarding ````AbstractFlowPass. It looks like the logic there requires an adjustment since the behavior of IsAutoPropertyWithGetAccessor``` has changed. We can no longer guarantee that accessing the property under the checked conditions is always equivalent to accessing a backing field. For example, if the corresponding accessor has a body, we don't really know what it is doing.
- `_ = BackingField?.GetAttributes();` We probably want to skip this if there are no attributes to look at. Do not want to bind accessors for no good reason.

Some things may have been outdated since I last updated it.

Reply:
> Do you expect another error? What exactly?

I expect no error on assignment to the ```field```. "It is my field, I do with it whatever I'd like."

> I was recording things in a local text file, but started to move everything to PROTOTYPE comments for easier tracking.

PROTOTYPE comments are good for implementation related issue. However, this is more a design issue and should be brought to LDM at some point. I'd recommend adding a "Open issues" section to the speclet .md file. And record those there as well.